### PR TITLE
NumberInput Component

### PR DESCRIPTION
**Pull request by personnamedmike**
react-hook-form `useForm` gets passed into NumberInput, so all RHF methods can be accessed in parent components.

Linked to issue https://github.com/airswap/airswap-voter-rewards/issues/29

### CODE REVIEW

**Comment by personnamedmike** [comment]
NOTE: it might be easier to merge this branch after merging https://github.com/airswap/airswap-voter-rewards/pull/37/ with main

**Review by gpxl-dev — APPROVED** [review]
I think there is more we can do with this numberinput but it looks good for now!

**Pull request merged**
